File: WebModule/common/models/Subcategory.php
<?php

namespace common\models;

use Yii;

/**
 * This is the model class for table "subcategories".
 *
 * @property int $id
 * @property string $description
 * @property int $category_id
 * @property int $is_deleted
 *
 * @property Category $category
 * @property Item[] $items
 */
class Subcategory extends \yii\db\ActiveRecord
{
    /**
     * {@inheritdoc}
     */
    public static function tableName()
    {
        return 'subcategories';
    }

    /**
     * {@inheritdoc}
     */
    public function rules()
    {
        return [
            [['description', 'category_id'], 'required'],
            [['category_id', 'is_deleted'], 'integer'],
            [['description'], 'string', 'max' => 255],
            [['category_id'], 'exist', 'skipOnError' => true, 'targetClass' => Category::class, 'targetAttribute' => ['category_id' => 'id']],
        ];
    }

    /**
     * {@inheritdoc}
     */
    public function attributeLabels()
    {
        return [
            'id' => 'ID',
            'description' => 'Description',
            'category_id' => 'Category ID',
            'is_deleted' => 'Is Deleted',
        ];
    }

    /**
     * Gets query for [[Category]].
     *
     * @return \yii\db\ActiveQuery
     */
    public function getCategory()
    {
        return $this->hasOne(Category::class, ['id' => 'category_id']);
    }

    /**
     * Gets query for [[Items]].
     *
     * @return \yii\db\ActiveQuery
     */
    public function getItems()
    {
        return $this->hasMany(Item::class, ['subcategory_id' => 'id']);
    }
}

File: Mobile/app/src/main/java/com/example/gofuel/model/subcategory/Subcategory.java
package com.example.gofuel.model.subcategory;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import com.example.gofuel.model.category.Category;

@Entity(tableName = "subcategories")
public class Subcategory {
    @PrimaryKey
    private final int id;
    private String description;
    private Category category;

    public Subcategory(int id, String description, Category category) {
        this.id = id;
        this.description = description;
        this.category = category;
    }

    public int getId() {
        return id;
    }

    public String getDescription() {
        return description;
    }

    public Category getCategory() {
        return category;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setCategory(Category category) {
        this.category = category;
    }
}
